TCAT/Controllers/FavoritesViewController.swift

```diff
@@ -0,0 +1,149 @@
+        view.addSubview(editButton)
+    }
+
+    private func setupFavoritesCollectionView() {
+        let favoritesFlowLayout = UICollectionViewFlowLayout()
+        favoritesFlowLayout.minimumLineSpacing = 12.0
+        favoritesFlowLayout.minimumInteritemSpacing = 4.0
+
+        favoritesCollectionView = UICollectionView(frame: .zero, collectionViewLayout: favoritesFlowLayout)
+        favoritesCollectionView.delegate = self
+        favoritesCollectionView.dataSource = self
+        favoritesCollectionView.register(FavoritesCollectionViewCell.self, forCellWithReuseIdentifier: favoritesReuseIdentifier)
+        favoritesCollectionView.backgroundColor = .clear
+        view.addSubview(favoritesCollectionView)
+    }
+
+    func setupTab() {
+        tab.backgroundColor = Colors.metadataIcon
+        tab.layer.cornerRadius = tabSize.height / 2
+        tab.clipsToBounds = true
+        view.addSubview(tab)
+    }
+
+    private func setupConstraints() {
+
+        let tabTopInset: CGFloat = 6
+        let horizontalPadding = 16
+        let titleBarTopPadding = 21
+        let favoritesTopPadding = 24
+        let bottomPadding = 18
+
+
+        favoritesTitleLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(horizontalPadding)
+            make.top.equalToSuperview().inset(titleBarTopPadding)
+        }
+
+        editButton.snp.makeConstraints { make in
+            make.trailing.equalToSuperview().inset(horizontalPadding)
+            make.top.equalToSuperview().inset(titleBarTopPadding)
+        }
+
+        favoritesCollectionView.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(horizontalPadding)
+            make.trailing.equalToSuperview().inset(horizontalPadding).priority(.high)
+            make.top.equalTo(favoritesTitleLabel.snp.bottom).offset(favoritesTopPadding)
+            make.bottom.equalToSuperview().offset(-bottomPadding).priority(.high)
+        }
+
+        tab.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalToSuperview().inset(tabTopInset)
+            make.size.equalTo(tabSize)
+        }
+
+
+    }
+
+}
+
+extension FavoritesViewController: UICollectionViewDataSource, UICollectionViewDelegate {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == favoritesCollectionView {
+            return favorites.count
+        }
+        return 0
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == favoritesCollectionView {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: favoritesReuseIdentifier, for: indexPath) as! FavoritesCollectionViewCell
+            cell.configure(for: favorites[indexPath.row])
+            return cell
+        }
+        return UICollectionViewCell()
```

Comment:
are we planning to use multiple collection views here? if not, we don't need this check and can just do `collectionView.dequeue`